akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -264,39 +236,25 @@ private[remote] object InboundCompression {
 /**
  * INTERNAL API
  * Handles counting and detecting of heavy-hitters and compressing them via a table lookup.
+ *
+ * Access to this class must be externally synchronised (e.g. by accessing it from only Actors or a GraphStage etc).
  */
 private[remote] abstract class InboundCompression[T >: Null](
-  val system:       ActorSystem,
+  val log:          LoggingAdapter,
   val settings:     ArterySettings.Compression,
   originUid:        Long,
   inboundContext:   InboundContext,
-  val heavyHitters: TopHeavyHitters[T],
-  stopped:          AtomicBoolean) {
+  val heavyHitters: TopHeavyHitters[T]) {
 
-  val log = Logging(system, getClass)
+  private[this] var state: InboundCompression.State[T] = InboundCompression.State.empty
 
-  // FIXME InboundCompressions should be owned by the Decoder stage, and then doesn't have to be thread-safe
-  private[this] val state: AtomicReference[InboundCompression.State[T]] =
-    new AtomicReference(InboundCompression.State.empty)
   // We should not continue sending advertisements to an association that might be dead (not quarantined yet)
   @volatile private[this] var alive = true
-  private[this] val resendCount = new AtomicInteger
+  private[this] var resendCount = 0
 
   private[this] val cms = new CountMinSketch(16, 1024, System.currentTimeMillis().toInt)
 
-  log.debug("Initializing inbound compression for originUid [{}]", originUid)
-  val schedulerTask: Option[Cancellable] =
-    tableAdvertisementInterval match {
-      case d: FiniteDuration ⇒
-        Some(system.scheduler.schedule(d, d)(runNextTableAdvertisement)(system.dispatcher))
-      case _ ⇒
-        None
-    }
-
-  def close(): Unit = {
-    schedulerTask.foreach(_.cancel())
-    log.debug("Closed inbound compression for originUid [{}]", originUid)
-  }
+  log.debug("Initializing {} for originUid [{}]", Logging.simpleName(getClass), originUid)
```

Comment:
notable change since previous logs were confusing - you'd think you have 2 compression instances which sounds scary, but actually is true and ok – one for manifests and one for actor refs, how the log message is better at explaining that.